### PR TITLE
ci: changeset workflow

### PR DESCRIPTION
**Pull request by floroz**
Automates check for `.changeset/` files in PR

After merging, we need to add to "Required" steps in branch protection

### CODE REVIEW

**Comment by floroz** [comment]
Run without changeset added: https://github.com/Frontify/fondue/actions/runs/8423035268/job/23063734426?pr=1773

**Comment by floroz** [comment]
Run with changeset added: https://github.com/Frontify/fondue/actions/runs/8423143329/job/23064078188?pr=1773

**Review by noahwaldner — APPROVED** [review]

File: .github/workflows/changeset-check.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        runs-on: ubuntu-latest
+
+        steps:
+            - name: Checkout code
+              uses: actions/checkout@v2
+              with:
+                    fetch-depth: 0
+            - name: Check for changeset file
+              run: |
+                    if git diff --name-only origin/main..HEAD -- '.changeset/' | grep -qE '^.changeset/.*\.md'; then
```
Comment (SamuelAlev) [inline .github/workflows/changeset-check.yaml:21]:
```suggestion
                    if git diff --name-only --diff-filter=A origin/main HEAD '.changeset/**.md'; then
```
Then we only check for "Added" files

File: .github/workflows/changeset-check.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+                        echo "Changeset file found!"
```
Comment (SamuelAlev) [inline .github/workflows/changeset-check.yaml:22]:
```suggestion
                        echo "Changeset file found!"
                        exit 0
```

File: .github/workflows/changeset-check.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+on:
+    pull_request:
+        branches:
+            - main
+        paths:
+            - 'packages/**'
+
+jobs:
+    changeset-check:
+        runs-on: ubuntu-latest
```
Comment (SamuelAlev) [inline .github/workflows/changeset-check.yaml:12]:
```suggestion
        runs-on: ubuntu-latest
        timeout: 3
```

File: .github/workflows/changeset-check.yaml (hunk 1 above)
Comment (floroz) [reply .github/workflows/changeset-check.yaml]: Implemented your suggestion but now we don't get a correct failure and have a false positive: https://github.com/Frontify/fondue/actions/runs/8433112659/job/23093496088?pr=1773

Comment (floroz) [reply .github/workflows/changeset-check.yaml]: fixed, we were missing to pipe `grep` to read the output of `git diff`

**Review by SamuelAlev — APPROVED** [review]

**Pull request merged**